extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -175,6 +176,7 @@ void build(SmallRyeHealthRecorder recorder,
 
     @BuildStep
     public void defineHealthRoutes(BuildProducer<RouteBuildItem> routes,
+            BuildProducer<OpenAPIBuildItem> openAPIProducer,
```

Comment:
Granted I don't know if this would work.

But could the spi dependency be `optional` and then use a different build step method that is only "used" if OpenAPI extension is actually present?

Reply:
I am not sure how to do this. asking on zulip...

Reply:
@kenfinnigan - not sure if it's possible to do this at the moment. I think with some changes to `BuildItem` we can maybe do something like `@BuildItem(ifInluded=Capability.SMALLRYE_OPENAPI)`, but I don't want to make that change as part of this PR or without consulting first. What I did in the mean time is to add SmallRye OpenAPI to the capabilities and do a normal `isPresent` against it when deciding to add the endpoint. Let me know what you think

Reply:
I have not looked at the code, sorry. But I have one question regarding potential future additions of the same kind: Will other Quarkus extensions be able to contribute additional entries to the OpenAPI document for the application without requiring changes to the OpenAPI extension. While I don't regard this as a requirement, I think it would be nice to have or at least for the design to allow for this evolution.

Reply:
Yip, other extensions can now just use the build item to contribute

Reply:
Oh, I see. You did end up moving the config flag to the health extension. This looks very much like what I was hoping for 👌